compiler/rustc_borrowck/src/diagnostics/conflict_errors.rs

```diff
@@ -602,6 +729,24 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
         true
     }
 
+    fn suggest_cloning(&self, err: &mut Diagnostic, ty: Ty<'tcx>, span: Span) {
+        let tcx = self.infcx.tcx;
+        // Try to find predicates on *generic params* that would allow copying `ty`
+        let infcx = tcx.infer_ctxt().build();
+        let clone_did = tcx.lang_items().clone_trait().unwrap();
+        let params = ty::List::empty();
+        let ty = tcx.erase_regions(ty);
+        let env = self.param_env;
```

Comment:
nit: can you inline these variables? It doesn't particularly improve readability to do, e.g., `let env = self.param_env`.